src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -966,7 +945,7 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
   /** Let's share a lot more about why we crash all over the place.
    *  People will be very grateful.
    */
-  protected var lastSeenContext: analyzer.Context = null
+  protected var lastSeenContext: analyzer.Context = analyzer.NoContext
```

Comment:
https://github.com/scala/scala/blob/v2.12.4/src/compiler/scala/tools/reflect/ToolBoxFactory.scala#L63 should be updated too

Reply:
Good catch.